sky/catalog/data_fetchers/fetch_runpod.py

```diff
@@ -0,0 +1,466 @@
+"""A script that generates the Runpod catalog.
+
+Usage:
+    python fetch_runpod.py [-h] [--output-dir OUTPUT_DIR] [--gpu-ids GPU_IDS]
+
+The RUNPOD_API_KEY environment variable must be set with a valid read-access
+RunPod API key.
+
+If --gpu-ids is provided, only fetches details for
+the specified GPU IDs (comma-separated). Otherwise, fetches all available GPUs.
+This flag is intended for testing and debugging individual GPU configurations.
+"""
+
+import argparse
+import os
+import sys
+import traceback
+from typing import Any, Dict, List, Optional, Union
+
+import pandas as pd
+import runpod
+from runpod.api import graphql
+
+# The API currently returns a dynamic number of vCPUs per pod that
+# changes frequently (less than 30 mins)
+# Therefore we hard code a default number of vCPUs from:
+# 1. The previous catalog, if the GPU exists there
+# 2. Or if not, the pricing page https://www.runpod.io/pricing
+# 3. Otherwise, the minimum of the returned# vCPU count from the API
+# The max count of GPUs per pod is set to 8 apart from A40 at 10
+DEFAULT_MAX_GPUS = 8
+DEFAULT_GPU_INFO: Dict[str, Dict[str, Union[int, float]]] = {
+    'A100-80GB': {
+        'vcpus': 8.0,
```

Comment:
should be 12, but generally points to something that would be difficult to maintain. These numbers are hard-coded but if I were to implement this I'd probably try to reduce hard-coding as much as possible. It's likely better to decide on an imperfect source of truth (the api or parsing the html pricing page), and leave it at that.